newIDE/app/src/MainFrame/Preferences/PreferencesProvider.js

```diff
@@ -317,6 +320,33 @@ export default class PreferencesProvider extends React.Component<Props, State> {
     );
   }
 
+  _getRecentFiles() {
+    const { values } = this.state;
+    const recentFiles = values.recentFiles
+    if(values.recentFiles) {
+      return values.recentFiles;
+    }
+    return null;
+  }
+
+  _setRecentFiles(data: FileMetadata) {
+    const { values } = this.state;
+    if (values.recentFiles.length > 5) values.recentFiles.pop();
+    if(!values.recentFiles.some(item => item.fileIdentifier === data.fileIdentifier)) {
+      this.setState({
+    values: {
+      ...values,
+      recentFiles: [
+        data,
+        ...values.recentFiles,
+      ]
+    }
+  },
+  () => this._persistValuesToLocalStorage(this.state)
+  );
+    }
+  }
+
```

Comment:
I've added these two functions which are basically setters and getters. `recentFiles` is an array of `fileMetadata` now, so we'll have the entire object rather than `fileIndentifier` as you suggested.